### PR TITLE
Feature/update TrajectoryFileInfo interface to support v2 data format

### PR DESCRIPTION
**Pull request by schoinh**
Resolves: [Update simularium-viewer interfaces for data format v2](https://aicsjira.corp.alleninstitute.org/browse/AGENTVIZ-1360)

This PR needs to get merged before I can make the time/space unit-related changes to simularium-website.

simularium-website uses the simularium-viewer interface `TrajectoryFileInfo` to check the shape of the .simularium file metadata coming into the UI. Since the .simularium file format was recently updated to v2, the interface needs to be updated as well. For now we want to preserve support for both formats, so instead of simply updating the interface, I split it into `TrajectoryFileInfoV1` and `TrajectoryFileInfoV2` interfaces. 

The key file to look at is src/simularium/types.ts. All other changes are just replacing the uses of the generic `TrajectoryFileInfo` with `TrajectoryFileInfoV1 | TrajectoryFileInfoV2`.

**Pull request recommendations:**

- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-##], adds tiff file format support_
- [x] Provide description and context of changes.
- [ ] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

**Review by blairlyons — APPROVED** [review]

**Review by meganrm — APPROVED** [review]

File: src/controller/index.ts [hunk 1]
```diff
@@ -63,7 +66,9 @@ export default class SimulariumController {
         };
 
         /* eslint-disable */
-        this.handleTrajectoryInfo = (msg: TrajectoryFileInfo) => {
+        this.handleTrajectoryInfo = (
```
Comment (toloudis) [inline src/controller/index.ts:69]: just curious, is the eslint-disable still needed here?

File: src/simularium/VisData.ts [hunk 2]
```diff
@@ -362,7 +368,10 @@ class VisData {
         this._dragAndDropFileInfo = fileInfo;
     }
 
-    public get dragAndDropFileInfo(): TrajectoryFileInfo | null {
+    public get dragAndDropFileInfo():
+        | TrajectoryFileInfoV1
+        | TrajectoryFileInfoV2
```
[the remark this answers is not in the document]
Comment (toloudis) [reply src/simularium/VisData.ts:373]: like (pseudocode):
`export typedef TrajectoryFileInfo = TrajectoryFileInfoV1 | TrajectoryFileInfoV2 ...`
in types.ts
and then just use TrajectoryFileInfo everywhere else?  
or (better yet) try to make only one place inthe code that can take either type (like when the message data arrives from the server), and always convert to the latest type.

**Review by EricJIsaac — APPROVED** [review]

File: src/controller/index.ts (hunk 1 above)
Comment (schoinh) [reply src/controller/index.ts:69]: Yeah, I just tried removing it and we get a few warnings about unused arguments.

File: src/simularium/VisData.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (schoinh) [reply src/simularium/VisData.ts:373]: Yeah, I totally agree. I'll look into your first suggestion. As for your second suggestion, how would we do the conversion? Import simulariumio into this repo?

[the remark this answers is not in the document]
Comment (schoinh) [reply src/simularium/VisData.ts]: Ok, implemented suggestion #1 and we have much fewer code changes now. :)

**Review by toloudis — APPROVED** [review]

File: src/simularium/VisData.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (toloudis) [reply src/simularium/VisData.ts]: > Yeah, I totally agree. I'll look into your first suggestion. As for your second suggestion, how would we do the conversion? Import simulariumio into this repo?

Definitely not import simulariumio... I'm not sure how best to handle it yet but we could discuss in slack or dev mtg.  
Is the code that distinguishes spatialUnitFactorMeters from spatialUnits already written or is that still to be implemented?

[the remark this answers is not in the document]
Comment (schoinh) [reply src/simularium/VisData.ts]: > Is the code that distinguishes spatialUnitFactorMeters from spatialUnits already written or is that still to be implemented?

Do you mean in simularium-website? I started writing some stuff, mostly just looking at the file version number to determine what to work with.

**Pull request merged**